mungegithub/mungers/submit-queue.go

```diff
@@ -1269,6 +1271,46 @@ func (sq *SubmitQueue) servePriorityInfo(res http.ResponseWriter, req *http.Requ
 </ol> `))
 }
 
+func (sq *SubmitQueue) getHealthSVG() []byte {
+	sq.Lock()
+	defer sq.Unlock()
+	blocked := false
+	blockingJobs := make([]string, 0)
+	blocked = !sq.health.MergePossibleNow
+	if blocked {
+		for job, status := range sq.e2e.GetBuildStatus() {
+			if status.Status == "Not Stable" {
+				job = strings.Replace(job, "kubernetes-", "", -1)
+				blockingJobs = append(blockingJobs, job)
+			}
+		}
+	}
+	status := "running"
+	color := "brightgreen"
+	if blocked {
+		status = "blocked"
+		color = "red"
+		sort.Strings(blockingJobs)
+		n := len(blockingJobs)
+		if n > 0 {
+			if n > 3 {
+				n = 3
+			}
+			status += " by " + strings.Join(blockingJobs[:n], ", ")
+			if len(blockingJobs) > 3 {
+				status += "..."
+			}
+		}
+	}
```

Comment:
Up to you :)

Maybe I can add a busy version for people like me now that you've done the
hard work ;)

On Thu, Jun 16, 2016 at 3:33 PM, Ryan Hitchman <notifications@github.com>
wrote:

> In mungegithub/mungers/submit-queue.go
> <https://github.com/kubernetes/contrib/pull/1224#discussion_r67436078>:
>
> > +	color := "brightgreen"
> > +	if blocked {
> > +		status = "blocked"
> > +		color = "red"
> > +		sort.Strings(blockingJobs)
> > +		n := len(blockingJobs)
> > +		if n > 0 {
> > +			if n > 3 {
> > +				n = 3
> > +			}
> > +			status += " by " + strings.Join(blockingJobs[:n], ", ")
> > +			if len(blockingJobs) > 3 {
> > +				status += "..."
> > +			}
> > +		}
> > +	}
>
> Do people really care about which PR number is currently undergoing
> testing? I don't want this badge to be too busy.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1224/files/8f4990dc753036079817359e4dc0e14a225d13be#r67436078>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAnglhkaXoxXmC5lKUvfc5a25pkWZAydks5qMc8sgaJpZM4I33Bh>
> .
>